ygen/codegen.go

```diff
@@ -434,6 +422,52 @@ func processModules(yangFiles, includePaths []string, options yang.Options) ([]*
 	return entries, nil
 }
 
+// langAgnosticDefinitions find the set of directory and enumeration entities
+// that are mapped to objects within output code in a language agnostic manner.
+// It takes an input set of YANG schema files (yangFiles), and the paths that
+// are to be used to search for includes (includePaths). Returns a map, keyed
+// by a string path to the schema element, of entities that are to be mapped
+// to directories with the output code (structs in the case of Go, messages
+// for proto3), a map - keyed by schema path - of enumerated entities that
+// are to have code generated for them, a ctree containing the schema tree
+// of the input YANG schemas.
+func langAgnosticDefinitions(yangFiles, includePaths []string, cfg GeneratorConfig) (map[string]*yang.Entry, map[string]*yang.Entry, *ctree.Tree, []error) {
+	modules, errs := processModules(yangFiles, includePaths, cfg.YANGParseOptions)
+	if len(errs) > 0 {
+		return nil, nil, nil, errs
+	}
+
+	// Extract the entities that are eligible to have code generated for
+	// them from the modules that are provided as an argument.
+	dirs := make(map[string]*yang.Entry)
+	enums := make(map[string]*yang.Entry)
+	var rootElems []*yang.Entry
+	for _, module := range modules {
+		findMappableEntities(module, dirs, enums, cfg.ExcludeModules, cfg.CompressOCPaths)
+		if module != nil && module.Dir != nil {
```

Comment:
This was an error -- looks like rather than investigating why I ended up with nils here before, I'd just handled the NPE. The issue was making a slice the wrong length.

Reply:
Added code to treat this as an error.